auslib/test/blobs/test_base.py

```diff
@@ -63,3 +67,132 @@ def testSchemaCaching(self):
             blob.validate('fake', [])
 
             self.assertEquals(yaml_load.call_count, 1)
+
+
+# Things we consider to be useful values in testing blobs. Basically, these
+# are things would actually show up in real blobs. Nesting is handled
+# further down.
+# Ideally we'd include st.boolean() here, but without a way to exclude
+# booleans from list items (which doesn't happen in the real world), we end
+# up with test failures because things like "1 in [True]" are True, which
+# ends up failing tests.
+useful_values = st.none() | st.floats(allow_nan=False) | st.text()
+
+
+@given(st.lists(useful_values), st.lists(useful_values), st.lists(useful_values))
+def test_merge_lists_unique_items_present(base, left_additional, right_additional):
+    left = deepcopy(base) + left_additional
+    right = deepcopy(base) + right_additional
+    expected = sorted(list(set(deepcopy(base) + left_additional + right_additional)))
+    got = sorted(merge_lists(base, left, right))
+    assert got == expected
+
+
+@given(st.lists(useful_values), st.lists(useful_values))
+def test_merge_lists_no_dupes(base, additional):
+    left = deepcopy(base) + additional
+    right = deepcopy(base) + additional
+    expected = sorted(list(set(deepcopy(base) + additional)))
+    got = sorted(merge_lists(base, left, right))
+    assert got == expected
+
+
+def unique_items_only(i):
+    """Returns a hashable value for various types to be used for testing
+    uniqueness when generating test data."""
+    if isinstance(i, dict):
+        return tuple(i.keys())
+    if isinstance(i, list):
+        return tuple(i)
```

Comment:
Not technically unique values in the list case, unless you cast it into a set first ?

Reply:
It depends on how you define unique :). In this case, we're looking at whether the list as a whole is unique compared to another list (aka: do we have the same items in the same places).